### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by jonwiseman**
- Changes to layout: Scripts --> backend, with code split into library code and testing code
- bot_controller.py moved to base level
- Added an additional field to the configuration file for testing  
- Incorporated build system
- Added SQL dump for build system
- Changes to SQL add and edit functions: return added/edited row rather than whole table  
- Added unit testing for most SQL functions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![CodeFactor](https://www.codefactor.io/repository/github/jonwiseman/lfj/badge)](https://www.codefactor.io/repository/github/jonwiseman/lfj)
+[![CodeFactor](https://www.codefactor.io/repository/github/jonwiseman/lfj/badge)](https://www.codefactor.io/repository/github/jonwiseman/lfj)  [![buddy pipeline](https://app.buddy.works/bowlian/lfj/pipelines/pipeline/249265/badge.svg?token=ab2372452e6e3f91c0b0f43ce4d2b5f58e6e232abaaf685199102b62422cc5d7 "buddy pipeline")](https://app.buddy.works/bowlian/lfj/pipelines/pipeline/249265)
```
Comment (jonwiseman) [inline README.md:1]: Added build passing badge

File: backend/lib/event_queries.py [hunk 2]
```diff
@@ -2,9 +2,9 @@
 import time
 from datetime import date
 from discord.ext import commands
-from game_queries import get_game_id
-from helper_commands import check_admin_status, AdminPermissionError
-from game_queries import GameNotFoundError
+from backend.lib.game_queries import get_game_id
+from backend.lib.helper_commands import check_admin_status, AdminPermissionError
+from backend.lib.game_queries import GameNotFoundError
```
Comment (jonwiseman) [inline backend/lib/event_queries.py:7]: Import changes due to unit testing and running bot_controller.py from project base

File: backend/lib/game_queries.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from discord.ext import commands
 from mysql.connector.errors import IntegrityError
-from helper_commands import check_admin_status, AdminPermissionError
+from backend.lib.helper_commands import check_admin_status, AdminPermissionError
```
Comment (jonwiseman) [inline backend/lib/game_queries.py:3]: another import change

File: backend/lib/game_queries.py [hunk 4]
```diff
@@ -37,6 +37,8 @@ def __init__(self, bot, cursor, cnx):
         """
         try:
             message = sql_delete_game(str(ctx.author), name, self.cursor, self.cnx)
+        except AdminPermissionError:
+            await ctx.send("Permission error: must be an admin to delete a game")
```
Comment (jonwiseman) [inline backend/lib/game_queries.py:41]: Added another error check

File: backend/lib/game_queries.py [hunk 5]
```diff
@@ -124,7 +130,7 @@ def sql_add_game(auth_user, game_id, name, cursor, cnx):
                    'values (%s, %s)', (game_id, name))  # add new user
     cnx.commit()  # commit changes to database
 
-    cursor.execute('select * from game')  # get new user table
+    cursor.execute('select * from game where name = %s', (name,))  # get new user table
```
Comment (jonwiseman) [inline backend/lib/game_queries.py:133]: Return only relevant row

File: backend/lib/game_queries.py [hunk 6]
```diff
@@ -152,7 +158,7 @@ def sql_edit_name(auth_user, old_name, new_name, cursor, cnx):
                    'set name = %s '
                    'where name = %s', (new_name, old_name))  # change the game table with new email
     cnx.commit()  # commit changes to user table
-    cursor.execute('select * from game')  # get new user table
+    cursor.execute('select * from game where name = %s', (new_name,))  # get new user table
```
Comment (jonwiseman) [inline backend/lib/game_queries.py:161]: Return only relevant row

File: backend/lib/helper_commands.py [hunk 7]
```diff
@@ -31,7 +31,7 @@ def check_admin_status(display_name, add, cursor):
     cursor.execute('select admin from user where display_name = %s', (display_name,))
     result = cursor.fetchall()
 
-    if add and (len(result) == 0 or result[0][0] == -1):        # adding to the database
+    if add and (len(result) == 0 or result[0][0] == 0):        # adding to the database
```
Comment (jonwiseman) [inline backend/lib/helper_commands.py:34]: Fixed error when adding users and checking permission.  There is no -1 flag in this table

File: backend/lib/user_queries.py [hunk 8]
```diff
@@ -1,6 +1,5 @@
 from discord.ext import commands
-from game_queries import get_game_id
-from helper_commands import check_admin_status, AdminPermissionError
+from backend.lib.helper_commands import check_admin_status, AdminPermissionError
```
Comment (jonwiseman) [inline backend/lib/user_queries.py:2]: More import changes

File: backend/tests/test_configuration.conf [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+username = root
+password = root
+host = mysql
+database = lfj
+
+[Testing]
+display_name = jon_wiseman#8494
+id = 8494
+email = wisemanj@etown.edu
+admin = 1
```
Comment (jonwiseman) [inline backend/tests/test_configuration.conf:11]: Configuration file for buddy.works testing

File: bot_controller.py [hunk 10]
```diff
@@ -1,18 +1,18 @@
...
 from discord.ext import commands
 import mysql.connector
-from user_queries import UserQueries
-from game_queries import GameQueries
-from event_queries import EventQueries
-from helper_commands import HelperCommands
+from backend.lib.user_queries import UserQueries
+from backend.lib.game_queries import GameQueries
+from backend.lib.event_queries import EventQueries
+from backend.lib.helper_commands import HelperCommands
```
Comment (jonwiseman) [inline bot_controller.py:7]: Even more import changes

File: bot_controller.py [hunk 11]
```diff
@@ -1,18 +1,18 @@
...
 
 
 def main():
     # event_channel = None
     # event_created = True
 
     config = configparser.ConfigParser()        # read and parse configuration file
-    config.read(r'../configuration.conf')
+    config.read(r'configuration.conf')
```
Comment (jonwiseman) [inline bot_controller.py:15]: Running from project root instead of the scripts folder

**Review by bowleri — APPROVED** [review]

**Review by MattGroho — APPROVED** [review]
Looks great!

**Pull request merged**